scripts/daq_utils.py

```diff
@@ -0,0 +1,226 @@
+                nodelist = " ".join(cols[5:])
+            else:
+                success = False
+            if success:
+                # Get logfile from job_id
+                scontrol_lines = call_subprocess(
+                    "scontrol", "show", "job", job_id
+                ).splitlines()
+                logfile = ""
+                for scontrol_line in scontrol_lines:
+                    if scontrol_line.find("StdOut") > -1:
+                        scontrol_cols = scontrol_line.split("=")
+                        logfile = scontrol_cols[1]
+
+                job_details[job_name] = {
+                    "job_id": job_id,
+                    "comment": comment,
+                    "state": state,
+                    "nodelist": nodelist,
+                    "logfile": logfile,
+                }
+        return job_details
+
+
+class DaqManager:
+    def __init__(self, opts):
+        self.opts = opts
+        self.verbose = False
+        for o, a in self.opts:
+            if o in ('-v', '--verbose'):
+                self.verbose = True
+        self.hutch = call_subprocess("get_info", "--gethutch")
+        self.user = self.hutch+'opr'
+        self.sbman = SbatchManager(self.user)
+        self.scripts_dir = f'/reg/g/pcds/dist/pds/{self.hutch}/scripts'
+        self.set_cnf_file_and_platform()
+
+    def isdaqbatch(self, quiet=False):
+        if self.hutch in DAQBATCH_HUTCHES:
+            if not quiet: print('true')
+            return True
+        else:
+            if not quiet: print('false')
+            return False
+
+    def isvaliduser(self):
+        if getpass.getuser().find('opr') > -1:
+            return True
+        return False
+
+    def waitfor(self, subcmd):
+        if subcmd == 'stop':
+            daq_host = self.wheredaq(quiet=True)
+            if daq_host is not None:
+                if self.isdaqbatch(quiet=True):
+                    for i_retry in range(MAX_RETRIES):
+                        daq_host = self.wheredaq(quiet=True)
+                        if daq_host is None:
+                            break
+                        print(f'wait for daqbatch to stop #retry: {i_retry}')
+                        time.sleep(1)
+                else:
+                    print(f'the DAQ did not stop properly (DAQ HOST: {daq_host}), exit now and try again or call your POC or the DAQ phone')
+
+    def set_cnf_file_and_platform(self):
+        self.cnf_file = f'{self.hutch}.py'
+        
+        cc = {'platform': None}
+        configfilename = os.path.join(self.scripts_dir, self.cnf_file)
+        try:
+            exec(compile(open(configfilename).read(), configfilename, 'exec'), {}, cc)
+            if type(cc['platform']) == type('') and cc['platform'].isdigit():
+                rv = int(cc['platform'])
+        except:
+            print('deduce_platform Error:', sys.exc_info()[1])
+            raise
+        self.platform = rv
```

Comment:
Small issues in the code here that I only noticed after reading other comments:
- `rv` is not guaranteed to be set, so it's possible to get us into a state where `self.platform = rv` raises a confusing `NameError`
- This is not the normal way to do type checking in Python, and this type check isn't necessary
- bare excepts are not great because they catch things like `KeyboardInterrupt`

I think instead we should lean on the built-in type checking from `int` and fail out into the decide platform error print if there's an issue.

```suggestion
            self.platform = int(cc['platform'])
        except Exception:
            print('deduce_platform Error:', sys.exc_info()[1])
            raise
```